src/python-common/ceph/cryptotools/select.py

```diff
@@ -0,0 +1,51 @@
+from typing import Dict
+
+import os
+
+from .caller import CryptoCaller
+
+
+_CC_ENV = 'CEPH_CRYPTOCALER'
+_CC_KEY = 'crypto_caller'
+_CC_REMOTE = 'remote'
+_CC_INTERNAL = 'internal'
+
+_CACHE: Dict[str, CryptoCaller] = {}
+
+
+def _check_name(name: str) -> None:
+    if name and name not in (_CC_REMOTE, _CC_INTERNAL):
+        raise ValueError(f'unexpected crypto caller name: {name}')
+
+
+def choose_crypto_caller(name: str = '') -> None:
+    _check_name(name)
+    if not name:
+        name = os.environ.get(_CC_ENV, '')
+        _check_name(name)
+    if not name:
+        name = _CC_REMOTE
```

Comment:
Is it not possible to attempt an internal import and fallback to remote?

Reply:
In any case, debug messages for these branches would be a good idea.

Reply:
Debug sounds good. Unfortunately, an attempt to import the module in the "wrong" subintepreter that *succeeds* will pollute the system. IOW, if we import the dashboard and it probes the cryptography module and it succeeds it will then break a subsequent import of cephadm. So it's not really "safe" to try to import it proactively.